mitodl/mit-learn
# Filter params for vector search

### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/6188

### Description (What does it do?)
This PR enables filter parameters on the vector search endpoint /api/v0/learning_resources_vector_search/

available filters:
- readable_id
- resource_type
- certification
- certification_type
- professional
- free
- course_feature
- content_feature_type
- run_id
- resource_id
- topic
- ocw_topic
- level
- department
- platform
- offered_by
- delivery
- resource_category

### How can this be tested?
1. checkout this branch
2. make sure you have learning resources loaded up locally
3. make sure you have embeddings generated otherwise regenerate them via ```python manage.py generate_embeddings --all --skip-contentfiles --recreate-collections```
4. once embeddings have been generated, visit the vector search endpoint /api/v0/learning_resources_vector_search/
5. try different filter parameters from the list above and validate the responses
6. play around with the limit and offset parameters and validate the results
